build: refactor cmake files

dundargoc opened the pull request:
- build: rely on default cmake installation if possible

The default cmake installation process for dependencies that use cmake
on all platforms is in general more robust and less verbose, so we rely
on that if possible.

- build(cmake): add modelines to enable syntax highlighting

- build: give example on complex regexes

This is just to allow the reader to get a quick understanding without
necessarily needing to know all the regex intricasies.

justinmk commented on cmake.deps/CMakeLists.txt:
```diff
@@ -146,6 +140,20 @@ endif()
 include(ExternalProject)
 set_directory_properties(PROPERTIES EP_PREFIX "${DEPS_BUILD_DIR}")
 
+set(GLOBAL_CMAKE_ARGS
```
if these are only used by deps, let's use the `DEPS_` prefix, which already has precedent.
```suggestion
set(DEPS_CMAKE_ARGS
```

dundargoc replied:
Good call.

dundargoc commented:
Apparently `CMAKE_ARGS` is ignored if `CONFIGURE_COMMAND` is specified, which essentially nullifies any benefits from this PR.

dundargoc committed:
build: rely on default cmake installation if possible

The default cmake installation process for dependencies that use cmake
on all platforms is in general more robust and less verbose, so we rely
on that if possible.

dundargoc committed:
build(cmake): add modelines to enable syntax highlighting

dundargoc committed:
build: give example on complex regexes

This is just to allow the reader to get a quick understanding without
necessarily needing to know all the regex intricasies.

justinmk commented on cmake.deps/cmake/LibiconvCMakeLists.txt:
```diff
@@ -95,3 +95,5 @@ install(TARGETS libcharset libiconv iconv
   PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+# vim: set ft=cmake:
```
good idea :)

justinmk reviewed (approved):
nice
